app/models/proposal.rb

```diff
@@ -107,18 +108,46 @@ def remove_approver(email)
     approval.destroy
   end
 
-  def initialize_approvals()
-    if self.linear? && self.approvals.any?
-      self.approvals.update_all(status: 'pending')
-      self.approvals.first.make_actionable!
-    elsif self.parallel?
-      self.approvals.update_all(status: 'actionable')
+  # Set the approver list, from any start state
+  def approvers=(approver_list)
+    approvals = approver_list.each_with_index.map do |approver, idx|
```

Comment:
WHO SAYS YOU'RE NOT A RUBYIST

![slow clap](http://media.giphy.com/media/nBn1PNOkkI3PW/giphy.gif)